This following patch was originally found at the debian mingw-w64 team's
binutils repo located here:
https://salsa.debian.org/mingw-w64-team/binutils-mingw-w64.git

Invoke the following in the aforementioned repo to see the original patch:

  $ git show da63f6b:debian/patches/reproducible-import-libraries.patch

Description: Make DLL import libraries reproducible
Author: Benjamin Moody <benjamin.moody@gmail.com>
Bug-Debian: https://bugs.debian.org/915055

--- a/ld/pe-dll.c
+++ b/ld/pe-dll.c
@@ -2844,6 +2844,7 @@
 
   bfd_set_format (outarch, bfd_archive);
   outarch->has_armap = 1;
+  outarch->flags |= BFD_DETERMINISTIC_OUTPUT;
 
   /* Work out a reasonable size of things to put onto one line.  */
   ar_head = make_head (outarch);
